spec/javascripts/tests/NewsletterSticky_spec.js

```diff
@@ -0,0 +1,221 @@
+describe('NewsletterSticky', function() {
+
+  'use strict';
+
+  beforeEach(function(done) {
+    var self = this;
+    requirejs(
+        ['jquery', 'NewsletterSticky', 'eventsWithPromises', 'utilities'],
+        function($, NewsletterSticky, eventsWithPromises, utilities) {
+          self.$html = $(window.__html__['spec/javascripts/fixtures/NewsletterSticky.html']).appendTo('body');
+          self.component = self.$html.find('[data-dough-component="NewsletterSticky"]');
+          self.NewsletterSticky = NewsletterSticky;
+          self.eventsWithPromises = eventsWithPromises;
+          self.eventsWithPromises.unsubscribeAll();
+
+          self.obj = new self.NewsletterSticky(self.component);
+          self.utilities = utilities;
+
+          self.clock = sinon.useFakeTimers();
+
+          done();
+
+        }, done);
+  });
+
+  afterEach(function() {
+    this.clock.restore();
+    this.$html.remove();
+    $(window).off('scroll');
+  });
+
+  describe('init', function() {
+    it('calls the right methods when initializing', function() {
+      var setListenersStub = sinon.stub(this.obj, "_setListeners"),
+          handleScrollStub = sinon.stub(this.obj, "_handleScroll"),
+          initialisedSuccessStub = sinon.stub(this.obj, "_initialisedSuccess");
+
+      this.obj.init();
+
+      expect(setListenersStub.callCount).to.be.equal(1);
+      expect(handleScrollStub.callCount).to.be.equal(1);
+      expect(initialisedSuccessStub.callCount).to.be.equal(1);
+    });
+  });
+
+  describe('_setListeners', function() {
+    beforeEach(function() {
+      this.obj._setVars();
+    });
+
+    it('calls _setScrollListener', function() {
+      var setScrollListenerStub = sinon.stub(this.obj, "_setScrollListener");
+
+      this.obj._setListeners(true);
+
+      expect(setScrollListenerStub.callCount).to.be.equal(1);
+    });
+
+    it('sets an event handle on the close button when true is passed in', function() {
+      var handleCloseClickStub = sinon.stub(this.obj, "_handleCloseClick");
+
+      this.obj._setListeners(true);
+
+      $('.news-signup-sticky__close').trigger('click');
+
+      expect(handleCloseClickStub.callCount).to.be.equal(1);
+    });
+
+    it('sets an event handle on the done button when true is passed in', function() {
+      var handleCloseClickStub = sinon.stub(this.obj, "_handleCloseClick");
+
+      this.obj._setListeners(true);
+
+      $('<button class="button--done"></button>').appendTo($('.news-signup-sticky'));
```

Comment:
It's all our code now :) but yes sounds cool, it hadn't really been reviewed before as it was still very work in progress